### ISSUE TITLE
Suggestion: Rename Item and Block to ItemKey and BlockKey

### ISSUE THREAD

**Issue by OroArmor** [QuiltMC/yarn#5]
This could help new users understand that items are not created each time, rather an ItemStack is. This could be done with documentation, but many new users do not know about documentation, so making the mappings clear should be a higher priority.

**Comment by LambdAurora** [comment]
I kinda disagree, they are definitely not keys, and it would be even more confusing.

`Block` and `Item` defines behavior of a block, item, etc. It represents the concept of block and item, to me the current names are accurate, could stretch and get to `BlockBehavior` and `ItemBehavior`

But I would like to note that, when MC is launched with the MC debug mode enabled, it will warn if a block or item is registered and does not end with `Block` or `Item`.

**Comment by Haven-King** [comment]
I agree with LambdAurora that `ItemKey` and `BlockKey` are not entirely accurate either. `ItemDefinition` and `BlockDefinition` would be the most apt names to me, as they define not just the behavior but also the settings for the item and block, respectively. That being said, I don't know that such a name change is necessary, but clarification in documentation would certainly be good.
